### PR TITLE
ERR: FutureWarning difficult to identify source

### LINKED ISSUE #26329: FutureWarning difficult to identify source
This is a suggestion, not a bug. The following resulted in a `FutureWarning`

    df = df.assign(block=(df.index.to_series().diff() != pd.Timedelta(15,'m')).cumsum())

> FutureWarning: The default of the 'keep_tz' keyword will change to True in a future release. You can set 'keep_tz=True' to obtain the future behaviour and silence this warning.

It wasn't in any way obvious why since the warning doesn't mention the method name and there's no stack trace (and futures warnings are only printed once). I had to search the source (which was also difficult as the message was split to maintain a character limit per line).

It would be nice if future warnings were more explicit, i.e. name the class and method as well as the argument. 

(In this case it's the `to_series()` method, which ideally wouldn't be required but `DateTimeIndex` doesn't yet implement diff())

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [x] closes #26329
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [n/a ] whatsnew entry

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -332,6 +332,7 @@ Timezones
 - Bug in :func:`to_datetime` where an uninformative ``RuntimeError`` was raised when passing a naive :class:`Timestamp` with datetime strings with mixed UTC offsets (:issue:`25978`)
 - Bug in :func:`to_datetime` with ``unit='ns'`` would drop timezone information from the parsed argument (:issue:`26168`)
 - Bug in :func:`DataFrame.join` where joining a timezone aware index with a timezone aware column would result in a column of ``NaN`` (:issue:`26335`)
+- FutureWarning message raised by :func:`DatetimeIndex.to_series` updated to identify source of warning (:issue:`26329`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: don't need a note for this

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.25.0.rst]: Meta-question: with review requests like this, is it better to just merge? I also thought that the note was unnecessary, but the "harm" of this note is small. Maybe better to just merge, and reduce CI burden / risk of a merge conflict coming up?

Comment (jreback) [reply doc/source/whatsnew/v0.25.0.rst]: i don’t think the ci burden is relevant here
we are pretty fast now adays
and better to fix things even small typos on the first go
as once things are merged it is almost impossible to see that something is amiss

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins keep em coming!

**Pull request merged**